autogpt_platform/frontend/src/app/(platform)/build/components/NewBlockMenu/BlockMenuSearchBar/useBlockMenuSearchBar.ts

```diff
@@ -0,0 +1,50 @@
+import { debounce } from "lodash";
+import { useEffect, useMemo, useRef, useState } from "react";
+import { useBlockMenuContext } from "../block-menu-provider";
+
+const SEARCH_DEBOUNCE_MS = 300;
+
+export const useBlockMenuSearchBar = () => {
+    const inputRef = useRef<HTMLInputElement>(null);
+    const [localQuery, setLocalQuery] = useState("");
+    const { setSearchQuery, setSearchId, searchId } = useBlockMenuContext();
+    
+    const searchIdRef = useRef(searchId);
+    useEffect(() => {
+      searchIdRef.current = searchId;
+    }, [searchId]);
+
+    const debouncedSetSearchQuery = useMemo(
+      () =>
+        debounce((value: string) => {
+          setSearchQuery(value);
+          if (value.length === 0) {
+            setSearchId(undefined);
+          } else if (!searchIdRef.current) {
+            setSearchId(crypto.randomUUID());
+          }
+        }, SEARCH_DEBOUNCE_MS),
+      [setSearchQuery, setSearchId],
+    );
```

Comment:
Is useMemo necessary here? I think we could simplify by removing it and just using debounce directly.